src/proj.h

```diff
@@ -180,16 +180,34 @@ struct PJ_INIT_INFO;
 typedef struct PJ_INIT_INFO PJ_INIT_INFO;
 
 /* Data types for list of operations, ellipsoids, datums and units used in PROJ.4 */
-struct PJ_LIST;
+struct PJ_LIST {
+    char    *id;                 /* projection keyword */
```

Comment:
would be nice if we could use const char*

Reply:
I don't see why we couldn't. Do you want to change the other `char *`s in `PJ_ELLPS` and friends as well?

Reply:
@rouault ping :-)

Reply:
I gave it a go. It wasn't as complicated as I feared. Do you mind taking another look at the updated commits @rouault ?